### PR TITLE
[Merged by Bors] - chore: rename iteratedDeriv_const_{smul,mul}

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Calculus/IteratedDeriv/Lemmas.lean [hunk 1]
```diff
@@ -65,15 +65,15 @@ theorem iteratedDerivWithin_const_sub (hn : 0 < n) (c : F) :
 @[deprecated (since := "2024-12-10")]
 alias iteratedDerivWithin_const_neg := iteratedDerivWithin_const_sub
 
-/-- Note: this is unrelated to `iteratedDeriv_const_smul`, where the scalar multiplication works on
-the domain. -/
+/-- Note: this is unrelated to `iteratedDeriv_comp_const_smul`, where the scalar multiplication
+works on the domain. -/
```
Comment (ADedecker) [inline Mathlib/Analysis/Calculus/IteratedDeriv/Lemmas.lean:69]: I would argue the renaming makes these unnecessary, right?

**Review by ADedecker — COMMENTED** [review]
Do you plan to add the "right" `iteratedDeriv_const_smul` statement in a later PR? And vice versa add the `within` version of `iteratedDeriv_comp_const_smul`? If not, could you add some `TODO`s?

**Comment by Ruben-VandeVelde** [comment]
> Do you plan to add the "right" `iteratedDeriv_const_smul` statement in a later PR?

Yes, after the next lean bump in early January (if this one gets in before then).

> And vice versa add the `within` version of `iteratedDeriv_comp_const_smul`?

If I can find the proof :)

> If not, could you add some `TODO`s?

Sure

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**